hosted/gke/k8s_chart_support/k8s_chart_support_suite_test.go

```diff
@@ -0,0 +1,257 @@
+package k8s_chart_support_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	"github.com/rancher-sandbox/ele-testhelpers/tools"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	nodestat "github.com/rancher/shepherd/extensions/nodes"
+	"github.com/rancher/shepherd/extensions/pipeline"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/shepherd/pkg/config"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+
+	"github.com/rancher/hosted-providers-e2e/hosted/gke/helper"
+	"github.com/rancher/hosted-providers-e2e/hosted/helpers"
+)
+
+var (
+	ctx                     helpers.Context
+	clusterName             string
+	k8sVersion              string
+	zone                    = helpers.GetGKEZone()
+	project                 = helpers.GetGKEProjectID()
+	rancherVersion          = os.Getenv("RANCHER_VERSION")
+	rancherUpgradedVersion  = os.Getenv("RANCHER_UPGRADE_VERSION")
+	kubeconfig              = os.Getenv("KUBECONFIG")
+	k8sUpgradedMinorVersion = os.Getenv("K8S_UPGRADE_MINOR_VERSION")
+)
+
+func TestK8sChartSupport(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "K8sChartSupport Suite")
+}
+
+var _ = BeforeSuite(func() {
+	Expect(rancherVersion).ToNot(BeEmpty())
+	// For upgrade tests, the rancher version should not be an unreleased version (for e.g. 2.8-head)
+	Expect(rancherVersion).ToNot(ContainSubstring("head"))
+
+	Expect(rancherUpgradedVersion).ToNot(BeEmpty())
+	Expect(k8sUpgradedMinorVersion).ToNot(BeEmpty())
+	Expect(kubeconfig).ToNot(BeEmpty())
+
+	By("Adding the necessary chart repos", func() {
+		err := kubectl.RunHelmBinaryWithCustomErr("repo", "add", catalog.RancherChartRepo, "https://charts.rancher.io")
+		Expect(err).To(BeNil())
+		err = kubectl.RunHelmBinaryWithCustomErr("repo", "add", fmt.Sprintf("rancher-%s", helpers.RancherChannel), fmt.Sprintf("https://releases.rancher.com/server-charts/%s", helpers.RancherChannel))
+		Expect(err).To(BeNil())
+	})
+
+	By(fmt.Sprintf("Installing Rancher Manager v%s", rancherVersion), func() {
+		helpers.DeployRancherManager(rancherVersion, true)
+	})
+
+})
+
+var _ = BeforeEach(func() {
+	var err error
+	ctx, err = helpers.CommonBeforeSuite(helpers.Provider)
+	Expect(err).To(BeNil())
+	clusterName = namegen.AppendRandomString(helpers.ClusterNamePrefix)
+
+	k8sVersion, err = helper.DefaultGKE(ctx.RancherClient, project, ctx.CloudCred.ID, zone, "")
+	Expect(err).To(BeNil())
+	GinkgoLogr.Info(fmt.Sprintf("Using GKE version %s", k8sVersion))
+
+})
+
+var _ = AfterEach(func() {
+	// The test must restore the env to its original state, so we install rancher back to its original version and uninstall the operator charts
```

Comment:
Yes. To restore rancher back to it's original version so that we can start a new test, and also to uninstall operator charts, otherwise the tests running in succession will not work.